esp-wifi/src/ble/mod.rs

```diff
@@ -96,3 +98,72 @@ impl HciOutCollector {
         &self.data[0..self.index]
     }
 }
+
+static BLE_HCI_READ_DATA: Mutex<RefCell<Vec<u8>>> = Mutex::new(RefCell::new(Vec::new()));
+
+#[derive(Debug, Clone)]
+pub struct ReceivedPacket {
+    pub data: Vec<u8>,
```

Comment:
This should be a `Box<[u8]>`, it's 4 bytes smaller.

Reply:
But we would then need to store the `len` again?

Reply:
No, why? Only the capacity is "missing", because the boxed slice can't grow.

Reply:
Ah yep you're right.